### DISCUSSION TITLE
Large numbers of users & Permissions

### DISCUSSION THREAD

**Discussion by ObaydaAlesawi** [spatie/laravel-permission#1738] [Q&A]
**I have the following users types:**
      1. Client (**50K**)
      2. Specialist
      3. Nova users(Admins)
Each one does have a separate guard and table.
My Questions are:
- Does it make sense to create a role for each user type and then assign this role to the users?
- Also is the package designed to handle this amount of users?

**Comment by drbyte** [comment]
Yes, roles are more performant than only assigning individual permissions to each user. It is also easier to update access for many users at once by using roles.

I've seen evidence of people using this package with very large numbers of users.
